agent/connect/ca/provider_vault_test.go

```diff
@@ -794,6 +794,98 @@ func TestVaultProvider_RotateAuthMethodToken(t *testing.T) {
 	}, 10*time.Second, 100*time.Millisecond)
 }
 
+func TestVaultProvider_ReconfigureIntermediateTTL(t *testing.T) {
+	SkipIfVaultNotPresent(t)
+
+	policy := `
```

Comment:
@kyhavlov : Should we modify the other tests in here to use the minimal, recommended permissions (demonstrated in this test) rather than the [maximal (not recommended) permissions](https://github.com/hashicorp/consul/blob/da8d6db30df512746c76acf0ca3773c8088ccfef/agent/connect/ca/provider_vault_test.go#L22-L30) currently used for most of the tests?

That would help us catch similar issues in the future, if the Vault CA provider code is modified to require access to a new endpoint (that needs new permissions). Without that change to the test permissions, it's easy for an issue like this one to recur.

Reply:
@kyhavlov : Should the read permissions on the `tune` path be included in the standard Vault policies for all tests? (I realize you'd need to temporarily remove that for the test case where you are seeing what happens if that isn't present, but it's now a part of the "standard" permissions we recommended, right?)

Reply:
Good call - none of the other tests need it yet, but it could be relevant in future tests.